---
description: Ошибка компилятора CS1017
title: Ошибка компилятора CS1017
ms.date: 07/20/2015
f1_keywords:
- CS1017
helpviewer_keywords:
- CS1017
ms.assetid: e0902e8a-b042-4711-a8a6-83456a3f88cb
ms.openlocfilehash: 9498a91c688412a05f031fc5f4ed9074324933cc
ms.sourcegitcommit: d579fb5e4b46745fd0f1f8874c94c6469ce58604
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 08/30/2020
ms.locfileid: "89142704"
---
# <a name="compiler-error-cs1017"></a><span data-ttu-id="878b0-103">Ошибка компилятора CS1017</span><span class="sxs-lookup"><span data-stu-id="878b0-103">Compiler Error CS1017</span></span>
<span data-ttu-id="878b0-104">Конструкции catch не могут использоваться после универсальной конструкции catch оператора try</span><span class="sxs-lookup"><span data-stu-id="878b0-104">Catch clauses cannot follow the general catch clause of a try statement</span></span>  
  
 <span data-ttu-id="878b0-105">Блок `catch` , не имеющий параметров, должен находиться в конце серии блоков `catch` .</span><span class="sxs-lookup"><span data-stu-id="878b0-105">A `catch` block that does not take any parameters must be the last in a series of `catch` blocks.</span></span> <span data-ttu-id="878b0-106">Дополнительные сведения об исключениях см. в разделе [исключения и обработка исключений](../programming-guide/exceptions/index.md).</span><span class="sxs-lookup"><span data-stu-id="878b0-106">For more information on exceptions, see [Exceptions and Exception Handling](../programming-guide/exceptions/index.md).</span></span>  
  
## <a name="example"></a><span data-ttu-id="878b0-107">Пример</span><span class="sxs-lookup"><span data-stu-id="878b0-107">Example</span></span>  
 <span data-ttu-id="878b0-108">Следующий пример приводит к возникновению ошибки CS1017:</span><span class="sxs-lookup"><span data-stu-id="878b0-108">The following sample generates CS1017:</span></span>  
  
```csharp  
// CS1017.cs  
using System;  
  
namespace x  
{  
    public class b : Exception  
    {  
    }  
  
    public class a  
    {  
        public static void Main()  
        {  
            try  
            {  
            }  
  
            catch   // CS1017, must be last catch  
            {  
            }  
  
            catch(b)  
            {  
                throw;  
            }  
        }  
    }  
}  
```
